### PR TITLE
Add `cocos2d::Layer` to MapController instance

### PR DESCRIPTION
**Pull request by stavenko**
Implemented https://github.com/TopSecretGames/game/issues/15
Also fit's requirements for https://github.com/TopSecretGames/game/issues/19

### CODE REVIEW

File: src/Classes/MapController.cpp [hunk 1]
```diff
@@ -2,35 +2,60 @@
...
+  listener->onTouchEnded = [&](cocos2d::Touch *touch,
+                               cocos2d::Event *event) { return true; };
+}
+
+void MapController::loadMap(std::string map) {
+  loadMapFromFile(map);
+  notifyListeners();
+}
+
+MapController::MapCollection MapController::getMaps() {
```
Comment (apocarteres) [inline src/Classes/MapController.cpp]: let's use get/set convention only for accessors and use find /  search / retrieve / fetch / lookup verbs for other exposing functions

File: src/Classes/MapController.h [hunk 2]
```diff
@@ -6,27 +6,32 @@
...
+  virtual void loadMapFromFile(const std::string &);
+
+  MapController(const MapController &);
+  MapController(cocos2d::Layer*);
+  MapController(cocos2d::Layer*, std::string);
+
+  MapCollection getMaps();
+  void loadMap(std::string);
+  void registerListener(IMapEventListener *);
+};
```
Comment (apocarteres) [inline src/Classes/MapController.h]: btw, why do we need:
getMaps(), loadMap() and loadMapFromFile() as public methods?
do we need CR on that?

Comment (stavenko) [inline src/Classes/MapController.h]: Actually I made them public as part of public interface. Except of `loadMapFromFile`. It needed for mocking. I'll look for possibility to make it private.

File: src/Classes/MapController.cpp (hunk 1 above)
Comment (stavenko) [inline src/Classes/MapController.cpp]: Agree, I'll rework this.

File: src/Classes/MapController.h (hunk 2 above)
Comment (apocarteres) [inline src/Classes/MapController.h]: Fine!

**Comment by stavenko** [comment]
This pull request goes fully to #17

**Pull request closed without merging**